accepted/0000-selective-versions-resolutions.md

```diff
@@ -0,0 +1,495 @@
+dependency `package-a` of `package-c`.
+
+## Resolutions
+
+Each sub-field of the `resolutions` field is called a *resolution*.
+It is a JSON field expressed by two strings: the package designation on the
+left and a version specification on the right.
+
+### Package designation
+
+A *resolution* contains on the left-hand side a glob pattern applied to
+the dependency tree (and not to the `node_modules` directory tree, since the
+latter is the result of yarn resolution being influenced by the *resolution*).
+
+- `a/b` denotes the directly nested dependency `b` of the project's
+dependency `a`.
+- `**/a/b` denotes the directly nested dependency `b`
+of all the dependencies and nested dependencies `a` of the project.
+- `a` denotes the project's dependency `a` (see below for a discussion on
+the matter of specifying a *resolution* for one of the project dependency).
+- `a/**/b` denotes all the nested dependencies `b` of the project's
+dependency `a`.
+- `**/a` denotes all the nested dependencies `a` of the project.
+- `a` is an alias for `**/a` (for retro-compatibility, see below, and because
+if it wasn't such an alias, it wouldn't mean anything as it would represent
+one of the non-nested project dependencies, which can't be overridden as
+explained below)
+- `**/a-*` denotes all the nested dependencies of the project whose
+name starts with `a-`.
+- `**` denotes all the nested dependencies of the project (a bad idea mostly,
+as well as all other designations ending with `**`).
+
+### Version specification
+
+A *resolution* contains on the right-hand side a version specification
+interpreted via the `semver` package as usually done in yarn.
+
+## Relation to non-nested dependencies
+
+The `devDependencies`, `optionalDependencies` and `dependencies` fields always
+take precedence over the
+`resolutions` field: if the user defines explicitly a dependency there,
+it means that he wants that version, even if it's specified with a non-exact
+specification. So the `resolutions` field only applies to nested-dependencies.
+Nevertheless, in case of incompatibility between the specification of a
+non-nested dependency version and a *resolution*, a warning is issued.
+
+This is coherent with the fact that the package designation `package-a` can be
+used safely as an alias of `**/package-a`: if it wasn't the case, `package-a`
+would designate one of the non-nested dependencies and would be ignored.
+
+## Retro compatibility for the `resolutions` field
+
+Until now, the`resolutions` field can contain *resolutions* of the following
+form (filled by `add --flat` or `install --flat`):
+```json
+  "resolutions": {
+    "package-a": "1.0.0"
+  }
+```
+
+With the current proposal, the package designation `package-a` is an alias for
+`**/package-a`: this means the behaviour of yarn with a project whose
+`resolutions` field contains *resolutions* filed by a pre-RFC yarn will be
+as expected: the nested dependencies will have the fixed version specified.

```

Comment:
Do you mean:
1. consider both syntaxes equally valid but prevent the user from mixing them in the same project, or
2. have a preference for the glob syntax and keep the non-glob syntax only for retro-compatibility purposes?

How would we achieve one or the other btw? Via printing a warning? For 2. I can easily see it happening, but for 1. it would seem strange I think...